src/shell/shell_misc.cpp

```diff
@@ -683,7 +683,7 @@ Bitu DOS_Shell::GetEnvCount() const
 	return num;
 }
 
-bool DOS_Shell::SetEnv(const char* entry, const char* new_string)
+bool DOS_Shell::SetEnv(std::string_view entry, std::string_view new_string)
```

Comment:
I guess string views are kinda OK in this case as you're treating them as a "universal type" that can accept both `const char*` and `std::string`, right? Like a unified interface.

Reply:
Yeah, it's just a double-edged swords, because if these string views are stored somewhere else, you can pull the rug from underneath them by freeing the original string they refer to. So I'd use them very, very cautiously. I guess in this case it's fine, though—but micro-optimisation.